Enhancement: Quick Ocarina

Dumbledork01 opened the pull request:
An enhancement that will allow the user to only have to pull out the ocarina to activate certain events. 
I believe most ocarina-triggered events are covered now.
(Note: The issue with the gate of time is due to me warping to the Temple of Time as child link without the spiritual stones and expecting the door to open. It's because I messed with a vanilla run, not the code itself.)

**Demonstration Video**
* https://youtu.be/JNMEIGoKY5o

Dumbledork01 committed:
Added Quick Ocarina

Logic for Quick Ocarina

jbodner09 commented:
Getting into Darunia requires Zelda's lullaby, and then Darunia himself responds to Saria's song.  There are also triforce marks in Spirit temple, the entrance to zora's domain, the water temple raising/lowering, I believe in ganon's trials, and of course the great fairies.  Saria also triggers skullkid and unblocks the way to forest temple as an adult.  

There are all the grottos that open to song of storms, as well as draining the well to begin with, and the oasis outside spirit.  If you're near a cow, epona's song will trigger milk (or items in rando with cowsanity turned on).  All of the song of time blocks would probably benefit from this, as well as the zora river frogs.

PurpleHato commented:
There's also the Skull kid check with Saria's Song and all the scarecrow places with well the scarecrow song

Dumbledork01 commented:
@purplehato Would quick ocarina benefit from summoning the scarecrow quickly if there already is a pull-request with an enhancement to quickly summon the scarecrow?

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Quick-Ocarina-Enhancement

Dumbledork01 committed:
More quick ocarina locations

Timeblocks now react to quick ocarina
Grottos react to quick ocarina
Cows react to quick ocarina
Skull kid quick ocarina logic
Treasure Check quick ocarina
Mido Quick Ocarina
Darunia quick ocarina

Song locations like triforces on ground, water temple water levels, shadow temple song locations, and others are now covered by quick ocarina.

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Quick-Ocarina-Enhancement

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Quick-Ocarina-Enhancement

Dumbledork01 committed:
Fixed typo

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Quick-Ocarina-Enhancement

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Quick-Ocarina-Enhancement

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Quick-Ocarina-Enhancement

aMannus commented:
@JakeEdvenson Do you want to push this forward still? It's sort of on us that this has been sitting around for this long (sorry about that), which lead to quite a lot of merge conflicts by now.

In case you want to update it, we can take a look at it afterwards and get the neccesary playtesting underway.

Dumbledork01 commented:
@aMannus Apologies for the late response. I'm going to go ahead and close this MR. I'm not sure when/if I'll pick up work again on this project (I started a new job and have been busy) so if I do come back I can just implement this on a new branch.
